pkg/test/framework/resource/flags.go

```diff
@@ -46,19 +46,36 @@ func SettingsFromCommandLine(testID string) (*Settings, error) {
 		return nil, err
 	}
 
+	if err = validate(s); err != nil {
+		return nil, err
+	}
+
+	return s, nil
+}
+
+// validate checks that user has not passed invalid flag combinations to test framework.
+func validate(s *Settings) error {
 	if s.FailOnDeprecation && s.NoCleanup {
-		return nil,
-			fmt.Errorf("checking for deprecation occurs at cleanup level, thus flags -istio.test.nocleanup and" +
-				" -istio.test.deprecation_failure must not be used at the same time")
+		return fmt.Errorf("checking for deprecation occurs at cleanup level, thus flags -istio.test.nocleanup and" +
+			" -istio.test.deprecation_failure must not be used at the same time")
 	}
 
-	if s.Revision != "" && s.Revisions != nil {
-		return nil,
-			fmt.Errorf("cannot use --istio.test.revision and --istio.test.revisions at the same time," +
+	if s.Revision != "" {
+		if s.Revisions != nil {
+			return fmt.Errorf("cannot use --istio.test.revision and --istio.test.revisions at the same time," +
 				" --istio.test.revisions will take precedence and --istio.test.revision will be ignored")
+		}
+		// use Revision as the sole revision in RevVerMap
+		s.Revisions = RevVerMap{
+			s.Revision: "",
+		}
 	}
 
-	return s, nil
+	if s.Compatibility && s.Revisions == nil {
```

Comment:
Seems like `Compatibility` is off by default. Will that break jobs that rely on the existing behavior?

Reply:
Great point... OSS has no jobs using it but I'll ask around in T&R tomorrow to see if others do.

Reply:
From  T&R seems like we're good.